api/graphql/reservation_units/reservation_unit_serializers.py

```diff
@@ -419,17 +490,131 @@ def validate_for_publish(self, data):
                 "minPersons can't be more than maxPersons"
             )
 
+        if not data.get("pricings", []):
+            raise serializers.ValidationError(
+                "pricings is required and must have one ACTIVE and one optional FUTURE pricing"
+            )
+
+    def validate_pricing_fields(self, data):
+        pricings = data.get("pricings", [])
+        if not pricings:
+            return
+
+        active_count = 0
+        future_count = 0
+        other_count = 0
+
+        for pricing in pricings:
+            if ReservationUnitPricingHelper.is_active(pricing):
+                active_count += 1
+                if pricing.get("begins") > date.today():
+                    raise serializers.ValidationError(
+                        "ACTIVE pricing must be in the past or today"
+                    )
+            elif ReservationUnitPricingHelper.is_future(pricing):
+                future_count += 1
+                if pricing.get("begins") <= date.today():
+                    raise serializers.ValidationError(
+                        "FUTURE pricing must be in the future"
+                    )
+            else:
+                other_count += 1
+
+        is_draft = data.get("is_draft", getattr(self.instance, "is_draft", False))
+        if (not is_draft and active_count != 1) or (is_draft and active_count > 1):
+            raise serializers.ValidationError(
+                "reservation unit must have exactly one ACTIVE pricing"
+            )
+        if future_count > 1:
+            raise serializers.ValidationError(
+                "reservation unit can have only one FUTURE pricing"
+            )
+        if other_count > 0:
+            raise serializers.ValidationError(
+                "only ACTIVE and FUTURE pricings can be mutated"
+            )
+
+    @staticmethod
+    def handle_pricings(pricings: List[Dict[Any, Any]], reservation_unit):
+        for pricing in pricings:
+            new_pricing = ReservationUnitPricing(
```

Comment:
You could just `ReservationUnitPricing.objects.create(**pricing, reservation_unit=reservation_unit)`

Reply:
Ah, of course! 🤦  Fixed 👍